src/main/kotlin/no/nav/familie/ef/sak/service/steg/BehandlingSteg.kt

```diff
@@ -0,0 +1,133 @@
+package no.nav.familie.ef.sak.service.steg
+
+import no.nav.familie.ef.sak.repository.domain.Behandling
+import no.nav.familie.ef.sak.repository.domain.BehandlingStatus
+import no.nav.familie.ef.sak.repository.domain.BehandlingType
+
+interface BehandlingSteg<T> {
+
+    fun utførStegOgAngiNeste(behandling: Behandling,
+                             data: T): StegType
+
+    fun stegType(): StegType
+
+
+    fun hentNesteSteg(behandling: Behandling): StegType {
+        return behandling.steg.hentNesteSteg(utførendeStegType = this.stegType(), behandlingType = behandling.type)
+    }
+
+    fun preValiderSteg(behandling: Behandling, stegService: StegService? = null) {}
+    fun postValiderSteg(behandling: Behandling) {}
+
+}
+
+enum class StegType(val rekkefølge: Int,
+                    val tillattFor: List<BehandlerRolle>,
+                    private val gyldigIKombinasjonMedStatus: List<BehandlingStatus>) {
+
+    REGISTRERE_SØKNAD(
```

Comment:
Skal systemet støtte papirsøknad i første omgang? Hvis vi bare skal støtte digital søknadsdialog: Kan vi ha et felles steg for å registrere søknad og personopplysninger? Det kan systemet gjøre automatisk :)

Reply:
Enig i at vi ikke trenger å tenke papirsøknad med en gang, men:
tenker du å rename denne til `REGISTRERE_OPPLYSNINGER` eller fjerne hele dette steget?

Reply:
Jeg tenkte på å rename det :)